frontend/src/App.test.tsx

```diff
@@ -1131,6 +1135,22 @@ describe("App.sendRerunBackMsg", () => {
 
 //   * handlePageNotFound has branching error messages depending on pageName
 describe("App.handlePageNotFound", () => {
+  let pushStateSpy: any
+
+  beforeEach(() => {
+    pushStateSpy = jest.spyOn(
+      window.history,
+      // @ts-ignore
+      "pushState"
+    )
+  })
+
+  afterEach(() => {
+    // @ts-ignore
+    pushStateSpy.mockRestore()
+    window.history.pushState({}, "", "/")
+  })
+
```

Comment:
Thanks for catching this!

If I'm remembering correctly, the tests that `window.history.pushState` gets called were previously put in place because the `handlePageNotFound` method did call `pushState` at one point. I later refactored the code so it wasn't necessary to do so, and removing those lines must have been missed because we weren't resetting mocks properly thus the tests didn't fail.

Now that we don't expect `handlePageNotFound` to call `pushState`, I think it's now fine to stop mocking it in this `describe` block altogether.

Reply:
Oh, right!
Sorry I don't know why I added these unused mocks... 😂